### PR TITLE
Implemented: usage of performFind for fetching picklist details (#85zry5kg0)

### LINKED ISSUE #120: Unable to open the picklist from the view picklist page
## Current behavior
The picklist are listed on the view picklist page but we are unable to open it and go to the details page. We are getting something went wrong error.

## Expected behavior
User should be able to open the picklist and see the details. 

## Can you handle fixing this bug by yourself?

- [ ] YES
- [ ] NO

## Environment details

- Browser: 
- OS: 
- Code Version:

### PR DESCRIPTION
**Pull request by k2maan**
### Related Issues

Closes #120 

### Short Description and Why It's Useful

Replaced the picklist API with performFind, this fixed the issue where details for picklist by users other than the currently logged-in user were not being fetched.

### Screenshots of Visual Changes before/after (If There Are Any)

[Screencast from 24-04-23 03:21:57 PM IST.webm](https://user-images.githubusercontent.com/59442907/233962939-176ca0b1-0aec-4a10-9ca3-8c40a56af790.webm)

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/DivanteLtd/vsf-capybara/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/components/Picklist-item.vue [hunk 1]
```diff
@@ -21,12 +21,9 @@ export default defineComponent({
...
   props: ['picklists'],
   methods: {
-    viewPicklist (picklist: any) {
-      this.store.dispatch('picklist/setCurrentPicklist', { id: picklist.picklistId }).then((resp) => {
-        if (resp.pickingItemList) {
-          this.router.push({ path: `/picklist-details/${picklist.picklistId}` });
-        }
-      });
+    async viewPicklist (picklist: any) {
+      await this.store.dispatch('picklist/setCurrentPicklist', { id: picklist.picklistId })
```
[the remark this answers is not in the document]
Comment (k2maan) [reply src/components/Picklist-item.vue:27]: Added sir

File: src/store/modules/picklist/actions.ts [hunk 2]
```diff
@@ -88,40 +88,46 @@ const actions: ActionTree<PicklistState, RootState> = {
...
-    if (current.picklist && current.pickingItemList && current.picklist.picklistId === payload.id) {
-      return {
-        'pickingItemList': current.pickingItemList
-      }
+    let resp;
+    const params = {
+      "inputFields": {
+        "picklistId": payload.id,
+      },
+      "fieldList": ["productId", "productName", "picklistId", "locationSeqId", "picklistBinId"],
```
[the remark this answers is not in the document]
Comment (k2maan) [reply src/store/modules/picklist/actions.ts]: Fixed

File: src/store/modules/picklist/actions.ts [hunk 3]
```diff
@@ -88,40 +88,46 @@ const actions: ActionTree<PicklistState, RootState> = {
...
-      resp.data.pickingItemList.map((picklist: any) => {
-        picklist.isChecked = false;
-      })
-      if (resp.status === 200 && resp.data.pickingItemList && !hasError(resp)) {
-        commit(types.PICKLIST_CURRENT_UPDATED, { current: resp.data })
+      resp = await PicklistService.getPicklist(params);
+      if (!hasError(resp) && resp.data.count) {
+        const pickingItemList = resp.data.docs.map((picklist: any) => ({ ...picklist, isChecked: false }))
+
+        commit(types.PICKLIST_CURRENT_UPDATED, { pickingItemList })
```
[the remark this answers is not in the document]
Comment (k2maan) [reply src/store/modules/picklist/actions.ts]: Made the change

File: src/store/modules/picklist/actions.ts [hunk 4]
```diff
@@ -88,40 +88,47 @@ const actions: ActionTree<PicklistState, RootState> = {
   /**
    * Set current picklist data
    */
-  async setCurrentPicklist ({ commit, state }, payload) {
-    let resp;
-
-    const current = state.current as any
+  async setCurrentPicklist({ commit, state }, payload) {
+    let current = state.current as any
+    if (current.pickingItemList && current.picklistId === payload.id) {
```
Comment (adityasharma7) [inline src/store/modules/picklist/actions.ts:93]:
```suggestion
    if (current.picklistId && current.picklistId === payload.id) {
```

File: src/store/modules/picklist/actions.ts [hunk 5]
```diff
@@ -88,40 +88,47 @@ const actions: ActionTree<PicklistState, RootState> = {
...
+      return current.pickingItemList
```
Comment (adityasharma7) [inline src/store/modules/picklist/actions.ts:94]:
```suggestion
      return current```

File: src/store/modules/picklist/actions.ts [hunk 6]
```diff
@@ -88,40 +88,47 @@ const actions: ActionTree<PicklistState, RootState> = {
...
           })
         );
+
         productIds = [...productIds]
         if (productIds.length) {
           this.dispatch('product/fetchProducts', { productIds })
         }
-        return resp.data;
+
+        return pickingItemList;
```
Comment (adityasharma7) [inline src/store/modules/picklist/actions.ts:126]:
```suggestion
        return current;
```

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**